test/resources/knativeserving.go

```diff
@@ -36,20 +38,78 @@ import (
 
 const (
 	interval = 10 * time.Second
-	timeout  = 2 * time.Minute
+	timeout  = 5 * time.Minute
 )
 
+// CreateKnativeServingReady polls the status of the KnativeServing called name
+// from client every `interval` until `inState` returns `true` indicating it
+// is done, returns an error or timeout.
+func CreateKnativeServingReady(t *testing.T, clients *test.KnativeServingAlphaClients, names test.ResourceNames,
+	inState func(s *v1alpha1.KnativeServing, err error) (bool, error)) (*v1alpha1.KnativeServing, error) {
+	svc, err := CreateKnativeServing(t, clients, names)
+	if err != nil {
+		return nil, err
+	}
+
+	span := logging.GetEmitableSpan(context.Background(), fmt.Sprintf("WaitForKnativeServingState/%s/%s", names.KnativeServing, "KnativeServingIsReady"))
+	defer span.End()
+
+	var lastState *v1alpha1.KnativeServing
+	waitErr := wait.PollImmediate(interval, timeout, func() (bool, error) {
+		lastState, err := clients.KnativeServings.Get(svc.Name, metav1.GetOptions{})
+		return inState(lastState, err)
+	})
+
+	if waitErr != nil {
+		return lastState, errors.Wrapf(waitErr, "knativeserving %s is not in desired state, got: %+v", names.KnativeServing, lastState)
+	}
+	return lastState, nil
+}
+
+func getDefaultKSSpec() *v1alpha1.KnativeServingSpec {
+	return &v1alpha1.KnativeServingSpec{
+		Config: map[string]map[string]string{
+			"defaults": {
```

Comment:
If we're going to include any config here, we should verify that the operator updates the correct configmap[s], so choose a value that wouldn't be a default.

Reply:
@jcrossley3  The problem I had here is that if I only defined the name and namespace, KS would never go to the ready status, and no deployment would be created.

Reply:
I believe setting a pointer to an empty spec should work. FWIW, we should make it that an `nil` pointer does the right thing (use defaults) as well. This is a bug if it doesn't.

Reply:
Tests pass for me locally after deleting `Spec: *getDefaultKSSpec(),` from the `KnativeServing` instance, so I think we're good. I still think it's useful to test with/without config and ensure the corresponding configmaps are updated, though.